File: unused/parse_publications.R
# Function to parse a single publication entry
parse_publication <- function(text) {
  # Initialize empty publication data frame row with specified columns in order
  pub <- data.frame(
    authors = character(1),
    doi = character(1),
    journal = character(1),
    pmid = character(1),
    title = character(1),
    year = character(1),
    link = character(1),
    stringsAsFactors = FALSE
  )
  
  # Extract year (assuming it's a 4-digit number)
  year_match <- regmatches(text, regexpr("\\b(19|20)\\d{2}\\b", text))
  if (length(year_match) > 0) {
    pub$year <- year_match
  }
  
  # Extract PMID (assuming it's an 8-digit number)
  pmid_match <- regmatches(text, regexpr("\\b\\d{8}\\b", text))
  if (length(pmid_match) > 0) {
    pub$pmid <- pmid_match
  }
  
  # Extract DOI if present
  doi_match <- regmatches(text, regexpr("10\\.\\d{4,9}/[-._;()/:A-Z0-9]+", text, ignore.case = TRUE))
  if (length(doi_match) > 0) {
    pub$doi <- doi_match
  }
  
  # Try to extract title (usually the first line)
  lines <- strsplit(text, "\n")[[1]]
  if (length(lines) > 0) {
    pub$title <- trimws(lines[1])
  }
  
  # Try to extract authors (usually after title and before journal)
  author_match <- regmatches(text, regexpr("([A-Z][a-z]+(?: [A-Z][a-z]+)*)(?:,|\\.)", text))
  if (length(author_match) > 0) {
    pub$authors <- author_match
  }
  
  # Try to extract journal (usually after authors and before year)
  journal_match <- regmatches(text, regexpr("\\.\\s*([A-Z][^.,]+(?: [A-Z][^.,]+)*)", text))
  if (length(journal_match) > 0) {
    pub$journal <- trimws(sub("^\\.\\s*", "", journal_match))
  }
  
  # Try to extract link (usually the last line)
  if (length(lines) > 0) {
    last_line <- trimws(lines[length(lines)])
    if (grepl("^https?://", last_line)) {
      pub$link <- last_line
    }
  }
  
  return(pub)
}

# Function to parse the entire publications file
parse_publications_file <- function(input_file) {
  # Read the input file
  text <- readLines(input_file, warn = FALSE)
  
  # Combine lines into a single string
  text <- paste(text, collapse = "\n")
  
  # Split into individual publications (assuming they're separated by blank lines)
  publications <- strsplit(text, "\n\\s*\n")[[1]]
  
  # Parse each publication
  parsed_pubs <- lapply(publications, parse_publication)
  
  # Combine into a single data frame
  result <- do.call(rbind, parsed_pubs)
  
  return(result)
}

# Function to write publications to tab-delimited file
write_publications_tab <- function(publications, output_file) {
  # Ensure the output file has .txt extension
  if (!grepl("\\.txt$", output_file)) {
    output_file <- paste0(output_file, ".txt")
  }
  
  # Write tab-delimited file
  write.table(publications, 
              file = output_file,
              sep = "\t",
              row.names = FALSE,
              quote = FALSE,
              na = "")
  
  cat(sprintf("Successfully wrote %d publications to %s\n", nrow(publications), output_file))
}

# Main function
main <- function() {
  # Get command line arguments
  args <- commandArgs(trailingOnly = TRUE)
  
  if (length(args) != 2) {
    cat("Usage: Rscript parse_publications.R input_file.txt output_file.txt\n")
    quit(status = 1)
  }
  
  input_file <- args[1]
  output_file <- args[2]
  
  # Parse publications
  publications <- parse_publications_file(input_file)
  
  # Write to tab-delimited file
  write_publications_tab(publications, output_file)
}

# Run the main function
main() 

File: unused/parse_publications.py
import re
import csv
import sys

def parse_publication(text):
    """Parse a single publication entry from text format to structured data."""
    # Initialize the publication dictionary
    pub = {
        'Title': '',
        'Authors': '',
        'Year': '',
        'Journal': '',
        'PMID': '',
        'Link': ''
    }
    
    # Extract year (assuming it's a 4-digit number)
    year_match = re.search(r'\b(19|20)\d{2}\b', text)
    if year_match:
        pub['Year'] = year_match.group()
    
    # Extract PMID (assuming it's an 8-digit number)
    pmid_match = re.search(r'\b\d{8}\b', text)
    if pmid_match:
        pub['PMID'] = pmid_match.group()
    
    # Try to extract title (usually the first line or before the authors)
    lines = text.split('\n')
    if lines:
        pub['Title'] = lines[0].strip()
    
    # Try to extract authors (usually after the title and before the journal)
    # This is a simple implementation - you might need to adjust based on your format
    author_match = re.search(r'([A-Z][a-z]+(?: [A-Z][a-z]+)*)(?:,|\.)', text)
    if author_match:
        pub['Authors'] = author_match.group(1)
    
    # Try to extract journal (usually after authors and before year)
    journal_match = re.search(r'\.\s*([A-Z][^.,]+(?: [A-Z][^.,]+)*)', text)
    if journal_match:
        pub['Journal'] = journal_match.group(1).strip()
    
    return pub

def parse_publications_file(input_file):
    """Parse a text file containing multiple publications."""
    publications = []
    
    try:
        with open(input_file, 'r', encoding='utf-8') as f:
            # Split the file into individual publications
            # This assumes publications are separated by blank lines
            raw_pubs = f.read().split('\n\n')
            
            for raw_pub in raw_pubs:
                if raw_pub.strip():  # Skip empty entries
                    pub = parse_publication(raw_pub)
                    publications.append(pub)
    
    except FileNotFoundError:
        print(f"Error: Could not find file {input_file}")
        sys.exit(1)
    
    return publications

def write_csv(publications, output_file):
    """Write publications to a CSV file."""
    fieldnames = ['Title', 'Authors', 'Year', 'Journal', 'PMID', 'Link']
    
    try:
        with open(output_file, 'w', newline='', encoding='utf-8') as f:
            writer = csv.DictWriter(f, fieldnames=fieldnames)
            writer.writeheader()
            writer.writerows(publications)
        print(f"Successfully wrote {len(publications)} publications to {output_file}")
    
    except IOError:
        print(f"Error: Could not write to file {output_file}")
        sys.exit(1)

def main():
    if len(sys.argv) != 3:
        print("Usage: python parse_publications.py input_file.txt output_file.csv")
        sys.exit(1)
    
    input_file = sys.argv[1]
    output_file = sys.argv[2]
    
    publications = parse_publications_file(input_file)
    write_csv(publications, output_file)

if __name__ == "__main__":
    main() 